File: android/app/src/main/java/com/stockswing/app/model/SignalResult.kt
package com.stockswing.app.model

import kotlinx.serialization.Serializable

@Serializable
/** 單支股票的選股結果（即時行情 + 策略訊號 + 命中 preset combo） */
data class SignalResult(
    val code: String,
    val name: String,
    val quote: RealtimeQuote,
    val signals: StrategySignals,
    // key = preset.key（如 "long3_lean"）, value = 命中的 combo 列表（如 ["B+K+R", "B+R"]）
    val hitPresets: Map<String, List<String>> = emptyMap(),
    val ma5:  Double = 0.0,
    val ma10: Double = 0.0,
    val ma20: Double = 0.0,
) {
    val totalComboHits: Int get() = hitPresets.values.sumOf { it.size }
}

File: ios/StockSwing/Models/SignalResult.swift
import Foundation

struct SignalResult: Codable, Identifiable {
    let code: String
    let name: String
    let quote: RealtimeQuote
    let signals: StrategySignals
    let hitPresets: [String: [String]]
    var ma5:     Double = 0
    var ma10:    Double = 0
    var ma20:    Double = 0
    var ma5Dir:    Int = 0   // 1=↑, -1=↓, 0=→
    var ma10Dir:   Int = 0
    var ma20Dir:   Int = 0
    // MA20 10日斜率趨勢：1=上升, -1=下降, 0=橫盤
    var ma20Trend: Int = 0

    var id: String { code }
    var totalComboHits: Int { hitPresets.values.reduce(0) { $0 + $1.count } }
}
